modules/sigmoidAnalyticsAdapter.js

```diff
@@ -0,0 +1,296 @@
+let bidWonConst = CONSTANTS.EVENTS.BID_WON;
+let bidRequestConst = CONSTANTS.EVENTS.BID_REQUESTED;
+let bidAdjustmentConst = CONSTANTS.EVENTS.BID_ADJUSTMENT;
+let bidResponseConst = CONSTANTS.EVENTS.BID_RESPONSE;
+
+let initOptions = { publisherIds: [], utmTagData: [], adUnits: [] };
+let bidWon = {options: {}, events: []};
+let eventStack = {options: {}, events: []};
+
+let auctionStatus = 'not_started';
+
+let localStoragePrefix = 'sigmoid_analytics_';
+let utmTags = ['utm_source', 'utm_medium', 'utm_campaign', 'utm_term', 'utm_content'];
+let utmTimeoutKey = 'utm_timeout';
+let utmTimeout = 60 * 60 * 1000;
+let sessionTimeout = 60 * 60 * 1000;
+let sessionIdStorageKey = 'session_id';
+let sessionTimeoutKey = 'session_timeout';
+
+function getParameterByName(param) {
+  let vars = {};
+  window.location.href.replace(location.hash, '').replace(
+    /[?&]+([^=&]+)=?([^&]*)?/gi,
+    function(m, key, value) {
+      vars[key] = value !== undefined ? value : '';
+    }
+  );
+
+  return vars[param] ? vars[param] : '';
+}
+
+function buildSessionIdLocalStorageKey() {
+  return localStoragePrefix.concat(sessionIdStorageKey);
+}
+
+function buildSessionIdTimeoutLocalStorageKey() {
+  return localStoragePrefix.concat(sessionTimeoutKey);
+}
+
+function updateSessionId() {
+  if (isSessionIdTimeoutExpired()) {
+    let newSessionId = utils.generateUUID();
+    localStorage.setItem(buildSessionIdLocalStorageKey(), newSessionId);
+  }
+  initOptions.sessionId = getSessionId();
+  updateSessionIdTimeout();
+}
+
+function updateSessionIdTimeout() {
+  localStorage.setItem(buildSessionIdTimeoutLocalStorageKey(), Date.now());
+}
+
+function isSessionIdTimeoutExpired() {
+  let cpmSessionTimestamp = localStorage.getItem(buildSessionIdTimeoutLocalStorageKey());
+  return Date.now() - cpmSessionTimestamp > sessionTimeout;
+}
+
+function getSessionId() {
+  return localStorage.getItem(buildSessionIdLocalStorageKey()) ? localStorage.getItem(buildSessionIdLocalStorageKey()) : '';
+}
+
+function updateUtmTimeout() {
+  localStorage.setItem(buildUtmLocalStorageTimeoutKey(), Date.now());
+}
+
+function isUtmTimeoutExpired() {
+  let utmTimestamp = localStorage.getItem(buildUtmLocalStorageTimeoutKey());
+  return (Date.now() - utmTimestamp) > utmTimeout;
+}
+
+function buildUtmLocalStorageTimeoutKey() {
+  return localStoragePrefix.concat(utmTimeoutKey);
+}
+
+function buildUtmLocalStorageKey(utmMarkKey) {
+  return localStoragePrefix.concat(utmMarkKey);
+}
+
+function checkOptions() {
+  if (typeof initOptions.publisherIds === 'undefined') {
+    return false;
+  }
+
+  return initOptions.publisherIds.length > 0;
+}
+
+function checkAdUnitConfig() {
+  if (typeof initOptions.adUnits === 'undefined') {
+    return false;
+  }
+
+  return initOptions.adUnits.length > 0;
+}
+
+function buildBidWon(eventType, args) {
+  bidWon.options = initOptions;
+  if (checkAdUnitConfig()) {
+    if (initOptions.adUnits.includes(args.adUnitCode)) {
```

Comment:
Use includes by importing it from core-js 
`import includes from 'core-js/library/fn/array/includes';`
Usage `includes(<array>, <item>`

`includes` doesn’t work in older versions of IE. The imported version is polyfilled to work with ie, rather than Babel doing any kind of transforms that might end up not working in target browsers